vms/platformvm/txs/txheap/heap.go

```diff
@@ -23,107 +20,57 @@ type Heap interface {
 }
 
 type heapTx struct {
-	tx    *txs.Tx
-	index int
-	age   int
+	tx  *txs.Tx
+	age int
 }
 
 type txHeap struct {
-	self heap.Interface
-
-	txIDToIndex map[ids.ID]int
-	txs         []*heapTx
-	currentAge  int
-}
-
-func (h *txHeap) initialize(self heap.Interface) {
-	h.self = self
-	h.txIDToIndex = make(map[ids.ID]int)
+	heap       heap.Map[ids.ID, heapTx]
+	currentAge int
 }
 
 func (h *txHeap) Add(tx *txs.Tx) {
-	heap.Push(h.self, tx)
+	txID := tx.ID()
+	if h.heap.Contains(txID) {
+		return
+	}
+	htx := heapTx{
+		tx:  tx,
+		age: h.currentAge,
+	}
+	h.currentAge++
+	h.heap.Push(txID, htx)
 }
 
 func (h *txHeap) Get(txID ids.ID) *txs.Tx {
-	index, exists := h.txIDToIndex[txID]
-	if !exists {
-		return nil
-	}
-	return h.txs[index].tx
+	got, _ := h.heap.Get(txID)
```

Comment:
I don't think `got` is guaranteed to be non-nil?

Reply:
`got` is a struct, not a pointer. So it's `zero` value will produce the expected result here.